src/Functions/array/arrayIndex.h

```diff
@@ -574,94 +624,247 @@ struct ArrayIndexGenericNullImpl
+            const ColumnNullable * const col_array_nested_nullable =
+                checkAndGetColumn<ColumnNullable>(&col_array->getData());
 
-            const auto & null_map2 = block.getByPosition(arguments[3]).column;
-            if (null_map2)
-                null_map_item = &assert_cast<const ColumnUInt8 &>(*null_map2).getData();
+            if (!col_array_nested_nullable)
+                return false;
+
+            col_array_nested_lc = checkAndGetColumn<ColumnLowCardinality>(
+                    &col_array_nested_nullable->getNestedColumn());
+
+            if (!col_array_nested_lc)
+                return false;
         }
 
-        const auto item_arg = block.getByPosition(arguments[1]).column.get();
+        auto col_res = ResultColumnType::create();
+        col_res->getData().resize_fill(col_array->getOffsets().size()); /// fill with default values
 
-        if (item_arg->onlyNull())
-            ArrayIndexNumNullImpl<T, IndexConv>::vector(col_nested->getData(), col_array->getOffsets(),
-                col_res->getData(), null_map_data);
-        else if (const auto item_arg_const = checkAndGetColumnConst<ColumnVector<U>>(item_arg))
-            ArrayIndexNumImpl<T, U, IndexConv>::vector(col_nested->getData(), col_array->getOffsets(),
-                item_arg_const->template getValue<U>(), col_res->getData(), null_map_data, nullptr);
-        else if (const auto item_arg_vector = checkAndGetColumn<ColumnVector<U>>(item_arg))
-            ArrayIndexNumImpl<T, U, IndexConv>::vector(col_nested->getData(), col_array->getOffsets(),
-                item_arg_vector->getData(), col_res->getData(), null_map_data, null_map_item);
-        else
-            return false;
+        const auto [null_map_data, null_map_item] = nullMapsBuilder(block, arguments); //null maps for outer Nullable.
+        const IColumn * col_arg = block.getByPosition(arguments[1]).column.get();
+
+        const size_t size = isColumnConst(*col_arg)
+            ? 1 /// We have a column with just one value. Arbitrary n is allowed (as the column is const, so take 0).
+            : col_arg->size();
+
+        for (size_t i = 0; i < size; ++i)
+        {
+            if (col_arg->onlyNull())
+            {
+                ArrayIndexNumNullImpl<
+                    ConcreteAction,
+                    /* already resized*/ false>::vector(
+                    col_array->getOffsets(),
+                    col_res->getData(),
+                    null_map_data);
+
+                continue;
+            }
+
+            const StringRef elem = col_arg->getDataAt(i);
+            const std::optional<UInt64> value_index = col_array_nested_lc->getDictionary().getOrFindIndex(elem);
+
+            if (!value_index)
+                continue; /// position already zeroed out
+
+            ArrayIndexNumImpl<
+                /* Initial data type -- DB::ReverseIndex index */ UInt64,
```

Comment:
This form of annotating template parameters looks alien to the codebase, I mostly see something like:
```
Type, /* comment */
```

rather than
```
/* comment */ Type, 
```